nemo_aligner/algorithms/ppo.py

```diff
@@ -406,46 +399,37 @@ def compute_rollout_metrics(self, rollout_batch):
     def run_validation(self):
         self.model.prepare_for_inference()
 
-        _, rollout_metrics, _ = self._run_inference(
-            self.val_dataloader_builder, consumed_samples=0, is_validation=True
-        )
+        _, rollout_metrics = self._run_inference(self.val_dataloader_builder, consumed_samples=0, is_validation=True)
 
         self.model.finish_inference()
         return rollout_metrics
 
     @torch.no_grad()
     def generate_rollouts(self):
-        timing_metrics = {}
+        with self.timer("prepare_for_inference"):
+            self.model.prepare_for_inference(timer=self.timer)
```

Comment:
i don't like passing in the timer here, it creates a coupling that is only there for metrics and not logic. I rather us just log this as prep for inference and assume we know that the steps > 1 are refit and not engine building

Reply:
I was trading off the complexity for making the metrics more granular and differentiating offloading adam states with refit measurements

Reply:
I don't think we should do this because it adds an argument that prepare for inference doesn't really need. if we want to fix this properly we should return timing metrics from prepare for inference. but that seems like a seperate PR

Reply:
reverted